docs/src/pages/components/autocomplete/Virtualize.tsx

```diff
@@ -46,16 +58,23 @@ const ListboxComponent = React.forwardRef<HTMLDivElement>(function ListboxCompon
   ref,
 ) {
   const { children, ...other } = props;
-  const itemData = React.Children.toArray(children);
+  const itemData: React.ReactChild[] = [];
+  (children as React.ReactChild[]).forEach(
```

Comment:
Would
```suggestion
  React.Children.forEach(children,
```
also work and avoid type casting?

Reply:
Yes, it would work but my understanding was that using React's inbuilt function was one of the major reasons for the slowdown. Will need to do some profiling - hopefully it will be alright.

Reply:
> Yes, it would work but my understanding was that using React's inbuilt function was one of the major reasons for the slowdown.

Can we make sure at the type-level that `children` are in fact an array?

Reply:
sorry for the late followup but I'm not sure by what do you me at the "type level" - do you have something like this in mind?

```
const ListboxComponent = React.forwardRef<HTMLDivElement, { children: React.ReactChild[] }>(function ListboxComponent(
  props,
  ref,
) {
  const { children, ...other } = props;
  const itemData: React.ReactChild[] = [];
  (children).forEach(
    (item: React.ReactChild & { children?: React.ReactChild[] }) => {
      itemData.push(item);
      itemData.push(...(item.children || []));
    },
  );
```

Reply:
following up on this @eps1lon, if you could have a look at my previous comment.

Reply:
> sorry for the late followup but I'm not sure by what do you mean by "at the type level" - do you have something like this in mind?

So that we can statically verify that `children` is already an array and don't need to use `React.Children`.

Reply:
we can specify that by defining the `ListBoxComponent` component like this like this
>const ListboxComponent = React.forwardRef<HTMLDivElement, { children: React.ReactChild[] }>(function ListboxComponent(
  props,
  ref,
) {}

this way we don't need to typecast it using `as React.Children` like before, if you're ok with this approach i can go ahead and commit it.